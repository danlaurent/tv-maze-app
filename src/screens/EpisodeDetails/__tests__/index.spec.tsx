import React from 'react';
import { render, RenderAPI, fireEvent } from '@testing-library/react-native';
import EpisodeDetailsScreen from '..';

const navigationMock: any = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

const routeMock: any = {
  params: {
    episode: {
      name: 'test',
      season: 1,
      number: 1,
      rating: {
        average: 10,
      },
      summary: 'A test episode',
    },
  },
};

describe('EpisodeDetailsScreen', () => {
  let tree: RenderAPI;
  beforeEach(() => {
    tree = render(
      <EpisodeDetailsScreen navigation={navigationMock} route={routeMock} />
    );
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const snapshot = tree.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  describe('When the back button is pressed', () => {
    beforeEach(async () => {
      await fireEvent.press(
        tree.getByTestId('EpisodeDetailsHeader_backButton')
      );
    });
    it('fires goBack action', () => {
      expect(navigationMock.goBack).toBeCalledTimes(1);
    });
  });
});
