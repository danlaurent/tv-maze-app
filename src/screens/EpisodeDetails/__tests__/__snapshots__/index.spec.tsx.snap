// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeDetailsScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
      }
    }
    testID="EpisodeDetailsHeader"
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "height": null,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 20,
          "paddingVertical": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="EpisodeDetailsHeader_backButton"
      >
        <Text />
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingHorizontal": 20,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "aspectRatio": 1.5,
              "width": 750,
            }
          }
        />
        <View
          style={
            Object {
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 24,
                  "fontWeight": "700",
                  "marginBottom": 8,
                  "marginTop": 16,
                }
              }
            >
              test
            </Text>
            <Text
              style={
                Object {
                  "color": "#6d6f7e",
                  "marginBottom": 8,
                  "textAlign": "center",
                }
              }
              testID="undefined_subtitle"
            >
              Season 1 | Episode 1
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#c6c7d2",
                    "fontSize": 16,
                    "fontWeight": "700",
                    "marginLeft": 8,
                  }
                }
                testID="undefined_rating"
              >
                10
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#6d6f7e",
              "textAlign": "center",
            }
          }
        >
          A test episode
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
